### ISSUE TITLE
def get_links_for_location - not liking more than 27 pic

### ISSUE THREAD

**Issue by dimahadgi** [InstaPy/InstaPy#1825]
Hi everyone!!

Guys, could someone help me with modifying like_util.py(def get_links_for_location)
Not liking more than 27 pic
I tried to fix it by my self but somthing go wrong and still 27 pic....

Help please =)
@uluQulu maybe you know what need to fix? Check pls. 
Thanks in advance!!!

**Comment by uluQulu** [comment]
Hi @dimahadgi modify your **like_util.py** file as _below_:

**1**. **import** `format_number` function _near_ other imports at top of the file (_if you have applied the fix from #1365, you most probably already have this import in there, check it and do not duplicate, they are the same_)
```python
from .util import format_number
```
**2**. **replace** your get_links_for_location function with this: [🌠](https://gist.github.com/uluQulu/f38ee4dfe06b5a1be3260ab5e9147d6c)
**3**. **add** this new function, `get_links`, to the _end_ of the **like_util.py** : [🛰](https://gist.github.com/uluQulu/54a2cf26345a05355c5c21c433bf6616) (_if you have applied the fix from #1365, you most probably already have this function in there, check it and do not duplicate, they are the same_) 

>I have not tested, if there is something wrong, please notify.

Good luck!

**Comment by dimahadgi** [comment]
Got this error:
```
D:\TEMP\InstaPy-master>set PYTHONIOENCODING=UTF-8

D:\TEMP\InstaPy-master>py quickstart.py

DevTools listening on ws://127.0.0.1:12524/devtools/browser/9e4eaf01-9111-4492-8ef8-2caa29ca27b2
INFO [2018-04-10 16:36:36] [dkhadzhy]  Session started - 2018-04-10 16:36:36
WARNING [2018-04-10 16:36:36] [dkhadzhy]  Sorry, Record Activity is not working on Windows. We're working to fix this soon!
[10136:4956:0410/163637.828:ERROR:upload_data_presenter.cc(73)] NOT IMPLEMENTED
INFO [2018-04-10 16:37:04] [dkhadzhy]  Logged in successfully!
[10136:4956:0410/163704.988:ERROR:upload_data_presenter.cc(73)] NOT IMPLEMENTED
INFO [2018-04-10 16:37:05] [dkhadzhy]  Location [1/1]
INFO [2018-04-10 16:37:05] [dkhadzhy]  --> b'225219858/dnipropetrovsk-ukraine/'
WARNING [2018-04-10 16:37:39] [dkhadzhy]  Too few images, skipping this location
INFO [2018-04-10 16:37:39] [dkhadzhy]  Liked: 0
INFO [2018-04-10 16:37:39] [dkhadzhy]  Already Liked: 0
INFO [2018-04-10 16:37:39] [dkhadzhy]  Inappropriate: 0
INFO [2018-04-10 16:37:39] [dkhadzhy]  Commented: 0
INFO [2018-04-10 16:37:39] [dkhadzhy]  Followed: 0
[8680:1364:0410/163739.697:ERROR:broker_win.cc(57)] Error reading broker pipe: ╩рэры с√ы чръЁ√Є. (0x6D)
INFO [2018-04-10 16:37:42] [dkhadzhy]  Session ended - 2018-04-10 16:37:42
INFO [2018-04-10 16:37:42] [dkhadzhy]  --------------------
```

**Comment by dimahadgi** [comment]
In quickstart.py I set: 
`session.like_by_locations(['225219858/dnipropetrovsk-ukraine/'], amount=50)`

**Comment by dimahadgi** [comment]
Tried with:

> 224442573/salton-sea/  

Got same error:
```
INFO [2018-04-10 16:44:53] [dkhadzhy]  --> b'224442573/salton-sea/'
WARNING [2018-04-10 16:45:27] [dkhadzhy]  Too few images, skipping this location
```

**Comment by filipemelo** [comment]
Is this problem still happening in the last version of instapy? 
Some time ago I had this issue, and stopped using it... I didn't have the time to fix in the past.
I will try to apply what uluQulu said.

**Comment by Edhim** [comment]
Had same error : 
**INFO [2018-04-10 17:33:27] [djedhim]  --> 217631504/casablanca-morocco/
WARNING [2018-04-10 17:34:00] [djedhim]  Too few images, skipping this location**

**Comment by uluQulu** [comment]
@dimahadgi 
find this line inside `like_by_locations` function of **instapy.py** file and **add** a new line after it: `raise`
```python
self.logger.warning('Too few images, skipping this location')   ##find this existing line
raise   ##add this new line
```
and output the full error in console after the last working logger message
so we will see what is making the problem, which element is not found...

**Comment by Edhim** [comment]
@uluQulu 
same error 
INFO [2018-04-10 23:27:15] [djedhim]  Session started - 2018-04-10 23:27:15
INFO [2018-04-10 23:27:44] [djedhim]  Logged in successfully!
INFO [2018-04-10 23:27:45] [djedhim]  Location [1/1]
INFO [2018-04-10 23:27:45] [djedhim]  --> 217631504/casablanca-morocco/
WARNING [2018-04-10 23:28:20] [djedhim]  Too few images, skipping this location
INFO [2018-04-10 23:28:20] [djedhim]  Liked: 0
INFO [2018-04-10 23:28:20] [djedhim]  Already Liked: 0
INFO [2018-04-10 23:28:20] [djedhim]  Inappropriate: 0
INFO [2018-04-10 23:28:20] [djedhim]  Commented: 0
INFO [2018-04-10 23:28:20] [djedhim]  Followed: 0
INFO [2018-04-10 23:28:20] [djedhim]  Session ended - 2018-04-10 23:28:20
INFO [2018-04-10 23:28:20] [djedhim]  --------------------

**Comment by uluQulu** [comment]
@Edhim,
I don't see any detail of the exception thrown (`NoSuchElementException`), have you added `raise` line as I mentioned in my previous comment?

**Comment by diarworld** [comment]
@uluQulu Ive added `raise` line ang got message:

  File "/InstaPy/instapy/instapy.py", line 579, in like_by_locations
    skip_top_posts)
  File "/InstaPy/instapy/like_util.py", line 673, in get_links_for_location
    "//span[contains(@class, '_fd86t')]").text)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 293, in find_element_by_xpath
    return self.find_element(by=By.XPATH, value=xpath)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 752, in find_element
    'value': value})['value']
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 236, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/errorhandler.py", line 192, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":"//span[contains(@class, '_fd86t')]"}
  (Session info: headless chrome=65.0.3325.181)
  (Driver info: chromedriver=2.37.544315 (730aa6a5fdba159ac9f4c1e8cbc59bf1b5ce12b7),platform=Linux 4.4.0-53-generic x86_64)

**Comment by uluQulu** [comment]
**That's** right @diarworld it's bcos I have never used like by locations feature, only now I see that the locations page does not have **possible** posts amount like it does in tags page...
So **in this case** you will **remove** all of these lines from `get_links_for_location` function in **like_util.py**:
```python
possible_posts = format_number(browser.find_element_by_xpath(
                            "//span[contains(@class, '_fd86t')]").text)

logger.info("desired amount: {}  |  top posts [{}]: {}  |  possible posts: {}".format(amount,
                                  ('enabled' if not skip_top_posts else 'disabled'), len(top_posts), possible_posts))
possible_posts = possible_posts if not skip_top_posts else possible_posts-len(top_posts)
amount = possible_posts if amount > possible_posts else amount
#sometimes pages do not have the correct amount of posts as it is written there, it may be cos of some posts is deleted but still keeps counted for the location
```
Having known the possible posts value gave us great flexibility in `get_links_for_tag` function, but for this, it is your job now, you must be more attentive to choose amounts for locations with enough pics, otherwise you will sometimes guess why it does get less pictures than desired...

> But it doesn't mean something undesired thing will happen, it is just that it would be so much better with that possible posts amount been there, I have also updated the file I have gisted above, feel free to reapply it

If there is another error happening, please notify!
>P.S.  By possible posts, I mean, e.g. "#sea 93,173,318 posts"

**Comment by dimahadgi** [comment]
@uluQulu **THANK YOU VERY MUCH!!! Everything works perfectly!!! =)**
So I attached fixed _instapy.py_ and _like_util.py_
_Just need to replace it with this one:_
[instapy.txt](https://github.com/timgrossmann/InstaPy/files/1897543/instapy.txt)
[like_util.txt](https://github.com/timgrossmann/InstaPy/files/1897544/like_util.txt)

**Comment by Edhim** [comment]
replaced the files and worked correctly - Thxxx

**Comment by timgrossmann** [comment]
@dimahadgi Are your fixes also merged into the core codebase here?

**Comment by dimahadgi** [comment]
Looks like merged =)

**Comment by uluQulu** [comment]
Hey @dimahadgi,
_regarding this situation: https://github.com/timgrossmann/InstaPy/issues/1365#issuecomment-386867146_;

I see that it slept less than `600` seconds (_somehow: either you've changed it **like_util.py** or slightly randomized sleep value took effect to be ~`570` secs_)
Now,
1. Make sure your sleep line is really `600`  (_BEWARE! The identical lines exist for `get_links_for_tag` function. But you will edit in `get_links_for_location` function._)
```python
logger.info("Cor! Did you send too many requests? ~ let's rest some")
sleep(600)   ## I mean this line
```
2. Try increasing that line a bit: _e.g._, put `655` or `700` for tests (_maybe even higher..._)
```python
logger.info("Cor! Did you send too many requests? ~ let's rest some")
sleep(700)   ## modify and test with higher sleep values
```
> After testing a few situations, if you get no change, please notify me. I will hopefully help.
_BTW, I have just tried the same location with the same amount (`1000`) and it did work fine returning all desired `1000` links and my **like_util.py** file is identical with the one shared in this page._
* And I see you are using `set_upper_follower_count` feature which is now deprecated, why don't you pull latest changes and  use more enhanced version of `set_relationship_bounds` with more functionalities? (_this is just a recommendation and does not affect the situation above_)

**Comment by dimahadgi** [comment]
Thanks @uluQulu !
Just set `sleep(700)` 
Will test and let you know!

**Comment by dimahadgi** [comment]
Tested for a couple day - all was fine.
Today got this one again:
```
D:\TEMP\InstaPy-master>set PYTHONIOENCODING=UTF-8

D:\TEMP\InstaPy-master>py quickstart.py

DevTools listening on ws://127.0.0.1:12980/devtools/browser/b78e26c8-fabc-4141-83fc-fd2641cebb6f
INFO [2018-05-21 10:02:36] [dkhadzhy]  Session started - 2018-05-21 10:02:36
WARNING [2018-05-21 10:02:36] [dkhadzhy]  Sorry, Record Activity is not working on Windows. We're working to fix this soon!
INFO [2018-05-21 10:03:04] [dkhadzhy]  Logged in successfully!
INFO [2018-05-21 10:03:06] [dkhadzhy]  Location [1/1]
INFO [2018-05-21 10:03:06] [dkhadzhy]  --> b'225219858/dnipropetrovsk-ukraine/'
INFO [2018-05-21 10:03:28] [dkhadzhy]  Insufficient amount of links ~ trying again: 1
INFO [2018-05-21 10:03:47] [dkhadzhy]  Insufficient amount of links ~ trying again: 2
INFO [2018-05-21 10:04:08] [dkhadzhy]  Insufficient amount of links ~ trying again: 3
INFO [2018-05-21 10:04:10] [dkhadzhy]  Cor! Did you send too many requests? ~ let's rest some
INFO [2018-05-21 10:11:55] [dkhadzhy]  Insufficient amount of links ~ trying again: 1
INFO [2018-05-21 10:12:13] [dkhadzhy]  Insufficient amount of links ~ trying again: 2
INFO [2018-05-21 10:12:37] [dkhadzhy]  Insufficient amount of links ~ trying again: 3
INFO [2018-05-21 10:12:39] [dkhadzhy]  '225219858/dnipropetrovsk-ukraine/' location POSSIBLY has less images than desired...
INFO [2018-05-21 10:12:43] [dkhadzhy]  [1/12]
INFO [2018-05-21 10:12:43] [dkhadzhy]  https://www.instagram.com/p/BjBnqXuFzBo/?taken-at=225219858
Unable to read VR Path Registry from C:\Users\╨Ф╨╕╨╝╨░\AppData\Local\openvr\openvrpaths.vrpath
[9968:2924:0521/101244.637:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:12:46] [dkhadzhy]  Image from: b'go.dnepr'
INFO [2018-05-21 10:12:51] [dkhadzhy]  Number of Followers: 59245
INFO [2018-05-21 10:12:51] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:12:51] [dkhadzhy]  [2/12]
INFO [2018-05-21 10:12:51] [dkhadzhy]  https://www.instagram.com/p/BjAfhDhF_59/?taken-at=225219858
INFO [2018-05-21 10:12:54] [dkhadzhy]  Image from: b'arenda_dnepr'
[9968:2924:0521/101256.478:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:12:58] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:12:58] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:12:58] [dkhadzhy]  [3/12]
INFO [2018-05-21 10:12:58] [dkhadzhy]  https://www.instagram.com/p/BjAcOs6FVCM/?taken-at=225219858
INFO [2018-05-21 10:13:00] [dkhadzhy]  Image from: b'arenda_dnepr'
INFO [2018-05-21 10:13:04] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:13:04] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:04] [dkhadzhy]  [4/12]
INFO [2018-05-21 10:13:04] [dkhadzhy]  https://www.instagram.com/p/BjAaVzAhGV3/?taken-at=225219858
[9968:2924:0521/101305.286:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:13:07] [dkhadzhy]  Image from: b'dreamway_dnepr_'
INFO [2018-05-21 10:13:11] [dkhadzhy]  Number of Followers: 8440
INFO [2018-05-21 10:13:11] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:11] [dkhadzhy]  [5/12]
INFO [2018-05-21 10:13:11] [dkhadzhy]  https://www.instagram.com/p/BjAYpRHl2D0/?taken-at=225219858
INFO [2018-05-21 10:13:13] [dkhadzhy]  Image from: b'arenda_dnepr'
[9968:2924:0521/101316.143:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:13:17] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:13:17] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:17] [dkhadzhy]  [6/12]
INFO [2018-05-21 10:13:17] [dkhadzhy]  https://www.instagram.com/p/BjAVya0FKiW/?taken-at=225219858
INFO [2018-05-21 10:13:21] [dkhadzhy]  Image from: b'arenda_dnepr'
INFO [2018-05-21 10:13:24] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:13:24] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:24] [dkhadzhy]  [7/12]
INFO [2018-05-21 10:13:24] [dkhadzhy]  https://www.instagram.com/p/BjAVelUFPN7/?taken-at=225219858
INFO [2018-05-21 10:13:28] [dkhadzhy]  Image from: b'arenda_dnepr'
[9968:2924:0521/101328.705:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:13:32] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:13:32] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:32] [dkhadzhy]  [8/12]
INFO [2018-05-21 10:13:32] [dkhadzhy]  https://www.instagram.com/p/BjAVRAvlH4x/?taken-at=225219858
INFO [2018-05-21 10:13:35] [dkhadzhy]  Image from: b'go.dnepr'
[9968:2924:0521/101338.207:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:13:39] [dkhadzhy]  Number of Followers: 59245
INFO [2018-05-21 10:13:39] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:13:39] [dkhadzhy]  [9/12]
INFO [2018-05-21 10:13:39] [dkhadzhy]  https://www.instagram.com/p/BjAS4uNn3zq/?taken-at=225219858
INFO [2018-05-21 10:13:42] [dkhadzhy]  Image from: b'fastar.kids'
INFO [2018-05-21 10:13:46] [dkhadzhy]  Number of Followers: 642
INFO [2018-05-21 10:13:46] [dkhadzhy]  Link: b'https://www.instagram.com/p/BjAS4uNn3zq/?taken-at=225219858'
INFO [2018-05-21 10:13:46] [dkhadzhy]  Description: b'\xf0\x9f\x92\x8e\xf0\x9f\x87\xba\xf0\x9f\x87\xa6\xd0\x90\xd0\xb1\xd0\xb5\xd1\x82\xd0\xba\xd0\xb0 \xd0\xa3\xd0\xba\xd1\x80\xd0\xb0\xd1\x97\xd0\xbd\xd1\x81\xd1\x8c\xd0\xba\xd0\xb0 \xd1\x83 3D \xd0\xb7 \xd0\xb4\xd0\xbe\xd0\xbf\xd0\xbe\xd0\xb2\xd0\xbd\xd0\xb5\xd0\xbd\xd0\xbe\xd1\x8e \xd1\x80\xd0\xb5\xd0\xb0\xd0\xbb\xd1\x8c\xd0\xbd\xd1\x96\xd1\x81\xd1\x82\xd1\x8e \xd1\x96 \xd0\xb3\xd0\xbe\xd0\xbb\xd0\xbe\xd1\x81\xd0\xbe\xd0\xbc\n.\n.\n\xf0\x9f\x92\x99\xd0\x97\xd0\xb0\xd0\xbf\xd1\x80\xd0\xbe\xd1\x88\xd1\x83\xd1\x94\xd0\xbc\xd0\xbe \xd0\xb4\xd0\xbe \xd1\x87\xd0\xb0\xd1\x80\xd1\x96\xd0\xb2\xd0\xbd\xd0\xbe\xd0\xb3\xd0\xbe \xd1\x81\xd0\xb2\xd1\x96\xd1\x82\xd1\x83 \xd0\xb4\xd0\xbe\xd0\xbf\xd0\xbe\xd0\xb2\xd0\xbd\xd0\xb5\xd0\xbd\xd0\xbe\xd1\x97 \xd1\x80\xd0\xb5\xd0\xb0\xd0\xbb\xd1\x8c\xd0\xbd\xd0\xbe\xd1\x81\xd1\x82\xd1\x96 \xd1\x96\xd0\xb7 \xd0\xb0\xd0\xb1\xd0\xb5\xd1\x82\xd0\xba\xd0\xbe\xd1\x8e \xd0\xb2\xd1\x96\xd0\xb4 FastAR Kids\xf0\x9f\x92\x8e\n\xf0\x9f\x93\x8d\xd0\x9c\xd0\xb8 \xd0\xb4\xd0\xbe\xd0\xbf\xd0\xbe\xd0\xbc\xd0\xbe\xd0\xb6\xd0\xb5\xd0\xbc\xd0\xbe \xd0\xb2\xd0\xb0\xd1\x88\xd0\xbe\xd0\xbc\xd1\x83 \xd0\xbc\xd0\xb0\xd0\xbb\xd1\x8e\xd0\xba\xd1\x83, \xd0\xbf\xd1\x96\xd0\xb7\xd0\xbd\xd0\xb0\xd0\xb2\xd0\xb0\xd1\x82\xd0\xb8 \xd0\xbd\xd0\xbe\xd0\xb2\xd0\xb5 \xd1\x82\xd0\xb0 \xd0\xb7\xd0\xb2\xd0\xb8\xd0\xba\xd0\xb0\xd1\x82\xd0\xb8 \xd0\xb4\xd0\xbe \xd0\xbd\xd0\xb0\xd0\xb2\xd1\x87\xd0\xb0\xd0\xbd\xd0\xbd\xd1\x8f, \xd1\x8f\xd1\x81\xd0\xba\xd1\x80\xd0\xb0\xd0\xb2\xd0\xbe, \xd1\x80\xd1\x96\xd0\xb7\xd0\xbd\xd0\xbe\xd0\xbc\xd0\xb0\xd0\xbd\xd1\x96\xd1\x82\xd0\xbd\xd0\xbe \xd1\x82\xd0\xb0 \xd0\xbd\xd0\xb0\xd0\xb4\xd0\xb7\xd0\xb2\xd0\xb8\xd1\x87\xd0\xb0\xd0\xb9\xd0\xbd\xd0\xbe \xd1\x86\xd1\x96\xd0\xba\xd0\xb0\xd0\xb2\xd0\xbe\xf0\x9f\x92\x99\n\xf0\x9f\x93\x8c\xd0\xa9\xd0\xbe\xd0\xb1 \xd0\xb7\xd0\xb0\xd0\xbc\xd0\xbe\xd0\xb2\xd0\xb8\xd1\x82\xd0\xb8 \xd0\xba\xd0\xbd\xd0\xb8\xd0\xb3\xd1\x83 \xd0\xbf\xd0\xb5\xd1\x80\xd1\x88\xd0\xb8\xd0\xbc\xd0\xb8 , \xd0\xbf\xd0\xb8\xd1\x88\xd1\x96\xd1\x82\xd1\x8c \xd0\xbd\xd0\xb0\xd0\xbc \xd0\xb2 Direct\xf0\x9f\x93\xa4\n.\n.\n.\n.\n.\n.\n.\n#vr #ar #augmented #reality #virtualreality #\xd0\xb4\xd0\xbe\xd0\xbf\xd0\xbe\xd0\xbb\xd0\xbd\xd0\xb5\xd0\xbd\xd0\xbd\xd0\xb0\xd1\x8f\xd1\x80\xd0\xb5\xd0\xb0\xd0\xbb\xd1\x8c\xd0\xbd\xd0\xbe\xd1\x81\xd1\x82\xd1\x8c #\xd0\xb0\xd0\xb1\xd0\xb5\xd1\x82\xd0\xba\xd0\xb0 #\xd0\xb4\xd0\xb5\xd1\x82\xd0\xb8\xc2\xa0 #\xd0\xba\xd0\xbd\xd0\xb8\xd0\xb3\xd0\xb8\xd0\xb4\xd0\xbb\xd1\x8f\xd0\xb4\xd1\x96\xd1\x82\xd0\xb5\xd0\xb9 #\xd0\xba\xd0\xbd\xd0\xb8\xd0\xb3\xd0\xb8\xd0\xb4\xd0\xbb\xd1\x8f\xd1\x80\xd0\xb0\xd0\xb7\xd0\xb2\xd0\xb8\xd1\x82\xd0\xb8\xd1\x8f\xc2\xa0 #\xd1\x80\xd0\xb0\xd0\xb7\xd0\xb2\xd0\xb8\xd1\x82\xd0\xb8\xd0\xb5\xd0\xb4\xd0\xb5\xd1\x82\xd0\xb5\xd0\xb9 #\xd1\x80\xd0\xb0\xd0\xb7\xd0\xb2\xd0\xb8\xd1\x82\xd0\xb8\xd0\xb5\xd1\x80\xd0\xb5\xd0\xb1\xd1\x91\xd0\xbd\xd0\xba\xd0\xb0'
INFO [2018-05-21 10:13:53] [dkhadzhy]  --> Image Liked!
INFO [2018-05-21 10:13:55] [dkhadzhy]  --> Not commented
INFO [2018-05-21 10:13:56] [dkhadzhy]  --> Not following
INFO [2018-05-21 10:13:57] [dkhadzhy]  [10/12]
INFO [2018-05-21 10:13:57] [dkhadzhy]  https://www.instagram.com/p/BjASOAmFtJb/?taken-at=225219858
INFO [2018-05-21 10:14:00] [dkhadzhy]  Image from: b'arenda_dnepr'
[9968:2924:0521/101401.379:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:14:04] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:14:04] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:14:04] [dkhadzhy]  [11/12]
INFO [2018-05-21 10:14:04] [dkhadzhy]  https://www.instagram.com/p/BjARPcxFVw_/?taken-at=225219858
INFO [2018-05-21 10:14:07] [dkhadzhy]  Image from: b'arenda_dnepr'
INFO [2018-05-21 10:14:12] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:14:12] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:14:12] [dkhadzhy]  [12/12]
INFO [2018-05-21 10:14:12] [dkhadzhy]  https://www.instagram.com/p/BjAO11Wl_2W/?taken-at=225219858
[9968:2924:0521/101412.670:ERROR:latency_info.cc(164)] Display::DrawAndSwap, LatencyInfo vector size 102 is too big.
INFO [2018-05-21 10:14:14] [dkhadzhy]  Image from: b'arenda_dnepr'
INFO [2018-05-21 10:14:18] [dkhadzhy]  Number of Followers: 49164
INFO [2018-05-21 10:14:18] [dkhadzhy]  --> Image not liked: Number of followers exceeds limit
INFO [2018-05-21 10:14:18] [dkhadzhy]  Liked: 1
INFO [2018-05-21 10:14:18] [dkhadzhy]  Already Liked: 0
INFO [2018-05-21 10:14:18] [dkhadzhy]  Inappropriate: 11
INFO [2018-05-21 10:14:18] [dkhadzhy]  Commented: 0
INFO [2018-05-21 10:14:18] [dkhadzhy]  Followed: 0
[1440:12200:0521/101418.625:ERROR:broker_win.cc(57)] Error reading broker pipe: ╩рэры с√ы чръЁ√Є. (0x6D)
INFO [2018-05-21 10:14:21] [dkhadzhy]  Session ended - 2018-05-21 10:14:21
INFO [2018-05-21 10:14:21] [dkhadzhy]  --------------------
```

**Comment by dimahadgi** [comment]
Started script again - got 192 images.

**Comment by uluQulu** [comment]
@dimahadgi 
It can be due to excessive request usage before starting that feature, so it stops at some point.  
Try to design your quickstart script to work with rests regularly and try to start some snacky features between scroller based features to avoid excessive requests to be sent.
What was the desired amount by the way?

**Comment by dimahadgi** [comment]
Sorry @uluQulu , but I do not understand what did you mean? 
I didn't ran anything before.
Amount of liked pic - 1000 likes
Here is my [quickstart.txt](https://github.com/timgrossmann/InstaPy/files/2022508/quickstart.txt)

By the way! Thanks for your advice to use "set_relationship_bounds"
I will use it, but little bit later, want to make "InstaPy" script running with no issues =)

**Comment by uluQulu** [comment]
Well @dimahadgi 
From the last log above,
```erlang
INFO [2018-05-21 10:04:10] [dkhadzhy]  Cor! Did you send too many requests? ~ let's rest some
INFO [2018-05-21 10:11:55] [dkhadzhy]  Insufficient amount of links ~ trying again: 1
```
it slept only 7 minutes.
_please re-read https://github.com/timgrossmann/InstaPy/issues/1825#issuecomment-386874343 attentively_
It must take a break at least for ten minutes in order to acquire new requests to the server.

>But if you have put it to sleep more than ten mins and it still sleeps less, then tell me to help you change the sleep logic in order to preserve accuracy.

**Comment by dimahadgi** [comment]
@uluQulu my fault!! 
Just updated Instapy by the latest version and apply your fix from #2042
For now working fine.
Sorry for wasting you time =)
Promise be more attentive!

**Comment by uluQulu** [comment]
@dimahadgi 
Glad to help :)
